ft_memccpy.c

```diff
@@ -0,0 +1,30 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_memccpy.c                                       :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: ekorley <marvin@42.fr>                     +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2016/09/22 16:24:48 by ekorley           #+#    #+#             */
+/*   Updated: 2016/09/22 16:36:45 by ekorley          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+
+void	*ft_memccpy(void *dest, const void *src, int c, size_t n)
+{
+	size_t					i;
+	unsigned char			*buffer;
+	const unsigned char		*source;
+
+	i = 0;
+	buffer = (unsigned char *)dest;
+	source = (const unsigned char *)src;
+	while (i < n && source[i] != c)
+	{
+		buffer[i] = source[i];
+		i++;
+	}
+	return (dest);
```

Comment:
Side effect implementation is great, return value is different than specified by the manual page for memccpy.  You are returning a pointer to the head of the destination right now, but you want to return something else:

- if you get through copying n bytes without encountering a c in the source, return a null pointer
- if you find a c in the source, copy that into the destination, and return a pointer to the NEXT byte